### PR TITLE
Flyway and sequenceID

### PR DESCRIPTION
**Pull request by JimmyWxzzz**

### CODE REVIEW

**Comment by JimmyWxzzz** [comment]
image

SequenceID

image

FlyWay

File: src/main/java/com/sweepy/urlService/urlServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    }
+
+
+
+    private void saveToRedis(String longUrl, String shortUrl, long time) {
+        System.out.println("outside");
+        redisService.setLTS(longUrl, shortUrl, time);
+        System.out.println("inlong");
+        redisService.setSTL(longUrl, shortUrl, time);
+        System.out.println("inshort");
```
Comment (zhangxyleon) [inline src/main/java/com/sweepy/urlService/urlServiceImpl.java:141]: should use Redis persistence to save  counter

File: src/main/java/com/sweepy/urlService/urlServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+        String shortInRedis = (String) redisService.get(longUrl);
+        if (shortInRedis == null) {
+            entry = urlRepository.findByLongUrl(longUrl);
+
+            if (!isValid(protocol + longUrl)) {
+                return "You must enter a valid long Url.";
+            } else if (entry == null || !Objects.equals(entry.getMethod(), method)) {
+                if (Objects.equals(method, "base62")) {
+                    shortUrl = base62Converter.encode(longUrl);
```
Comment (zhangxyleon) [inline src/main/java/com/sweepy/urlService/urlServiceImpl.java]: should use counter(sequence) to generate base62 code. In  base62convetr,  it just sums up long Url and converts the sum to base62.

File: src/main/java/com/sweepy/urlService/urlServiceImpl.java (hunk 1 above)
Comment (JimmyWxzzz) [reply src/main/java/com/sweepy/urlService/urlServiceImpl.java:141]: not sure what you mean, i have saved ID into redis in sequenceIdService.java

File: src/main/java/com/sweepy/urlService/urlServiceImpl.java (hunk 2 above)
Comment (JimmyWxzzz) [reply src/main/java/com/sweepy/urlService/urlServiceImpl.java]: addressed

File: src/main/java/com/sweepy/urlService/urlServiceImpl.java (hunk 1 above)
Comment (LeonYanc) [reply src/main/java/com/sweepy/urlService/urlServiceImpl.java:141]: Please talk to @zhangxyleon to get this comments resolved

**Review by LeonYanc — APPROVED** [review]

**Review by zhangxyleon — APPROVED** [review]

**Pull request merged**